messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/DefaultJMSConnectionFactoryResourceReferenceProcessor.java

```diff
@@ -47,8 +48,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
         if(deploymentUnit.getParent() == null) {
             final EEResourceReferenceProcessorRegistry eeResourceReferenceProcessorRegistry = deploymentUnit.getAttachment(Attachments.RESOURCE_REFERENCE_PROCESSOR_REGISTRY);
-            if(eeResourceReferenceProcessorRegistry != null) {
-                eeResourceReferenceProcessorRegistry.registerResourceReferenceProcessor(RESOURCE_REFERENCE_PROCESSOR);
+            if (eeResourceReferenceProcessorRegistry != null) {
+                final EEModuleDescription eeModuleDescription = deploymentUnit.getAttachment(Attachments.EE_MODULE_DESCRIPTION);
+                if (eeModuleDescription != null && eeModuleDescription.getDefaultResourceJndiNames().getDataSource() != null) {
```

Comment:
Should it be `getJmsConnectionFactory()` ?

Reply:
Yes, it should be getJmsConnectionFactory().

Reply:
fixed